ivy/array/device.py

```diff
@@ -1,8 +1,52 @@
 # global
 import abc
+from typing import Union, Optional, Any, Iterable
+
+import ivy
+
 
 # ToDo: implement all methods here as public instance methods
 
 
 class ArrayWithDevice(abc.ABC):
-    pass
+    def dev(self: ivy.Array) -> Union[ivy.Device, ivy.NativeDevice]:
+        """
+        ivy.Array instance method variant of ivy.device.dev. This method simply wraps
```

Comment:
we should not include the submodule: we should write `ivy.dev` rather than `ivy.device.dev`, the same is true for most other docstrings.